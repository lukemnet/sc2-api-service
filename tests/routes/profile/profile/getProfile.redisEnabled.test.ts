import fastify from 'fastify';
import fastifyRedis from 'fastify-redis-mock';
import server from '../../../../src/index';
import getConfig from '../../../helper';

describe('/profile/profile/:regionId/:realmId/:profileId (Redis enabled)', () => {
  const fastifyServer = fastify() as any;
  const url = '/profile/profile/1/1/1';
  const cacheSegment = 'profile-1-1-1';
  const expectedTTL = getConfig(true).redis.ttl.profile;
 
  beforeAll(async () => {
    fastifyServer.register(fastifyRedis, {
      host: '127.0.0.1',
      port: '6379',
      password: '',
      enableReadyCheck: true,
      dropBufferSupport: false,
    });
    fastifyServer.register(server, getConfig(true));
  });

  afterEach(() => {
    fastifyServer.close();
  });

  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.payload).toMatchSnapshot();
  });

  it('response is cached correctly', async () => {
    await fastifyServer.inject({ method: 'GET', url });
    const cachedResponse = await fastifyServer.redis.get(cacheSegment);
    expect(cachedResponse).toMatchSnapshot();
  });

  it('cached response has correct TTL', async () => {
    await fastifyServer.inject({ method: 'GET', url });
    const ttl = await fastifyServer.redis.ttl(cacheSegment);
    expect(ttl).toEqual(expectedTTL);
  });
});