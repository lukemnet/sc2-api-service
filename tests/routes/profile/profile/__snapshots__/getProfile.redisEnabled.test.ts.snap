// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/profile/profile/:regionId/:realmId/:profileId (Redis enabled) response is cached correctly 1`] = `"{\\"status\\":200,\\"data\\":{\\"summary\\":{\\"displayName\\":\\"sample profile object\\"}}}"`;

exports[`/profile/profile/:regionId/:realmId/:profileId (Redis enabled) returns correct response 1`] = `"{\\"status\\":200,\\"data\\":{\\"summary\\":{\\"id\\":\\"\\",\\"displayName\\":\\"sample profile object\\",\\"clanName\\":\\"\\",\\"clanTag\\":\\"\\",\\"portrait\\":\\"\\",\\"decalTerran\\":\\"\\",\\"decalProtoss\\":\\"\\",\\"decalZerg\\":\\"\\",\\"totalSwarmLevel\\":\\"\\",\\"totalAchievementPoints\\":\\"\\"}}}"`;
